activesupport/lib/active_support/subscriber.rb

```diff
@@ -58,23 +78,49 @@ def subscribers
         attr_reader :subscriber, :notifier, :namespace
 
         def add_event_subscriber(event) # :doc:
-          return if %w{ start finish }.include?(event.to_s)
+          return if is_invalid_event?(event.to_s)
 
-          pattern = "#{event}.#{namespace}"
+          pattern = prepare_pattern(event)
 
           # Don't add multiple subscribers (eg. if methods are redefined).
-          return if subscriber.patterns.include?(pattern)
+          return if is_pattern_subscribed?(pattern)
+
+          subscriber.patterns[pattern] = notifier.subscribe(pattern, subscriber)
+        end
+
+        def remove_event_subscriber(event) # :doc:
+          return if is_invalid_event?(event.to_s)
+
+          pattern = prepare_pattern(event)
+
+          return unless is_pattern_subscribed?(pattern)
+
+          notifier.unsubscribe(subscriber.patterns[pattern])
+          subscriber.patterns.delete(pattern)
+        end
+
+        def find_attached_subscriber(subscriber)
+          subscribers.find { |attached_subscriber| attached_subscriber.instance_of?(subscriber.class) }
+        end
+
+        def is_invalid_event?(event)
```

Comment:
Isn't `invalid_event?(event)` enough? IMHO it is not common to use `is_` (same as `get_`) as method prefixes in Ruby.

Reply:
ok I will update it. Thanks.